views.py

```diff
@@ -213,7 +143,8 @@ def view_profile():
 
     num_sightings = 0
     type_data = {}
-    
+
+    # Duplication indicates this section can be placed in its own utility function!
```

Comment:
This exact logic is in a route above. This is a good sign that you can extract this into it's own utility function.